### ISSUE TITLE
An annotation on an anonymous class creation is not checked against the declaration bound on JDK 11

### ISSUE THREAD

**Issue by aosen-xiong** [eisop/checker-framework#2009]
`new @Anno Class() {}` is validated against the declaration bound of the class being extended on
JDK 17 and later, but not on JDK 11, so the same source checks differently depending on the
compiler.

#### Reproducer

`framework/tests/viewpointtest/AnonymousClassBounds.java`:

```java
import viewpointtest.quals.*;

public class AnonymousClassBounds {
    @SuppressWarnings({"inconsistent.constructor.type", "super.invocation.invalid"})
    @A static class AClass {}

    void test() {
        new @A AClass() {};

        // @B is a sibling of @A, so it is outside AClass's declaration bound.
        // :: warning: (cast.unsafe.constructor.invocation)
        // :: error: (type.invalid.annotations.on.use)
        new @B AClass() {};

        // :: warning: (cast.unsafe.constructor.invocation)
        new @Bottom AClass() {};

        // :: error: (new.class.type.invalid)
        // :: error: (type.invalid.annotations.on.use)
        new @Top AClass() {};
    }
}
```

#### Expected

The same result on every supported JDK.

#### Actual

JDK 21 passes. JDK 11 (`ORG_GRADLE_PROJECT_useJdkVersion=11`) fails:

```
java.lang.AssertionError: 236 out of 238 expected diagnostics were found.
2 expected diagnostics were not found:
  AnonymousClassBounds.java:22: error: (type.invalid.annotations.on.use)
  AnonymousClassBounds.java:31: error: (type.invalid.annotations.on.use)
```

The same failure appears in the Mutability Checker as two missing
`type.invalid.annotations.on.use` in `checker/tests/pico-mutable-default/AnonymousClasses.java`.

**Comment by wmdietl** [comment]
#1909 previously tried making this more uniform, but apparently missed something.
